### PR TITLE
updates for multi-hash pixels

### PR DESCRIPTION
**Pull request by ferrants**
See: https://github.com/TraverseData/traverse-email-documentation/blob/patch-readme-updates/README.md#example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,64 +9,76 @@ Contents
...
 
-The Pixel Block is a set of 10 pixels, each of which redirects to a separate partner, passing them the recipient's hashed email address and allowing them to set a cookie.
+The Pixel Block is a set of at least 10 pixels, each of which redirects to a separate partner, passing the partner hashed versions of the recipient's email address and allowing the partner to set a cookie.
 
 Each pixel has the following form:
- 
->\<img border="0" width="1" height="1" src="http://`cname`/v1/`clientId`/`index`.gif?`hashType`=`hashValue`&domain=`domain`"/\>
+
+```
+<img border="0" width="1" height="1" src="http://{{cname}}/v1/{{clientId}}/{{index}}.gif?emailMd5Lower={{emailMd5Lower}}&emailSha1Lower={{emailSha1Lower}}&emailSha256Lower={{emailSha256Lower}}&domain={{domain}}"/\>
```
Comment (ferrants) [inline README.md]: this copy and pastes without the weird quotes that won't work in code

File: README.md [hunk 2]
```diff
@@ -11,64 +9,76 @@ Contents
...
-Some partners expect an MD5 hash, but others expect SHA-1. In order to maximize revenue, please provide both:
-
 | Parameter    | Description |
 | ------------ |------------ |
 | `emailMd5Lower` | MD5 hash of lowercased email address. |
 | `emailSha1Lower` | SHA1 hash of lowercased email address. |
+| `emailSha256Lower` | SHA256 hash of lowercased email address. |
 
-For example, if the email address is `foo@BAR.com`, then `emailMd5Upper` would be `6d881a14e8fb4886b2742f0b4aa10d30`.
+For example, if the email address is `foo@BAR.com`, then `emailMd5Lower` would be `f3ada405ce890b6f8204094deb12d8a8`.
```
Comment (ferrants) [inline README.md:54]: more common example

File: README.md [hunk 3]
```diff
@@ -11,64 +9,76 @@ Contents
...
+<img border="0" width="1" height="1" src="http://traverse.example.com/v1/YOUR-CLIENT-ID-HERE/7.gif?emailMd5Lower=f3ada405ce890b6f8204094deb12d8a8&emailSha1Lower=823776525776c8f23a87176c59d25759da7a52c4&emailSha256Lower=0c7e6a405862e402eb76a70f8a26fc732d07c32931e9fae9ab1582911d2e8a3b&domain=bar.com"/>
+<img border="0" width="1" height="1" src="http://traverse.example.com/v1/YOUR-CLIENT-ID-HERE/8.gif?emailMd5Lower=f3ada405ce890b6f8204094deb12d8a8&emailSha1Lower=823776525776c8f23a87176c59d25759da7a52c4&emailSha256Lower=0c7e6a405862e402eb76a70f8a26fc732d07c32931e9fae9ab1582911d2e8a3b&domain=bar.com"/>
+<img border="0" width="1" height="1" src="http://traverse.example.com/v1/YOUR-CLIENT-ID-HERE/9.gif?emailMd5Lower=f3ada405ce890b6f8204094deb12d8a8&emailSha1Lower=823776525776c8f23a87176c59d25759da7a52c4&emailSha256Lower=0c7e6a405862e402eb76a70f8a26fc732d07c32931e9fae9ab1582911d2e8a3b&domain=bar.com"/>
+```
 
+__Note__: We still support supplying one hash per pixel and repeating indexes for each hash type. This is not encouraged anymore. Providing all hashes available to each pixel will provide the most revenue. Example:
```
Comment (ferrants) [inline README.md]: I think it's important to indicate that this still works, as all of our clients have this way

File: README.md [hunk 4]
```diff
@@ -1,8 +1,6 @@
-Traverse Email Documentation
-----------------------------
+# Traverse Email Documentation v1.1:
```
Comment (ferrants) [inline README.md:1]: The container one had a version on it

Comment (this-sam) [reply README.md:1]: definitely a good practice.

File: README.md [hunk 5]
```diff
@@ -11,64 +9,76 @@ Contents
...
 | `clientId` | Your 36-character client ID (includes hyphens). | Yes |
 | `index` | How many pixels *with this hash type* came before this one? | Yes |
-| `hashType` | The [hash](#hashes) type. | Yes |
-| `hashValue` | The [hash](#hashes) value. | Yes |
+| `emailMd5Lower` | MD5 hash of lowercased email address. | No<sup>*</sup> |
+| `emailSha1Lower` | SHA1 hash of lowercased email address. | No<sup>*</sup> |
+| `emailSha256Lower` | SHA256 hash of lowercased email address. | No<sup>*</sup> |
 | `domain` | The recipient's domain. | No |
 
+<sup>*</sup> __Note__: At least one of the forms of hashes is required. Some partners expect an MD5 hash, but others expect SHA-1 or SHA-256. In order to maximize revenue, please provide all of `emailMd5Lower`, `emailSha1Lower`, and `emailSha256Lower`. If unable to provide all, it is important that you use the same hash parameters with each pixel.
```
Comment (this-sam) [inline README.md]: At least one of the hash types is required.

File: README.md (hunk 3 above)
Comment (this-sam) [reply README.md]: Agreed. I'm going to be pedantic about language, sorry. 

We continue to support supplying one hash per pixel and repeating indexes for each hash type. This is no longer the preferred method of integration. Providing all hashes available to each pixel allows Traverse to maximize your revenue.

Comment (ferrants) [reply README.md]: updated

**Comment by this-sam** [comment]
Just realized we load every pixel in the examples over http, when they're also available over https. Would you mind updating the URLs to http(s) ?

**Comment by ferrants** [comment]
updated to reference https

**Review by this-sam — APPROVED** [review]

**Pull request merged**